ftp-upload/src/lambda.ts

```diff
@@ -1,30 +1,36 @@
 import {Config} from './config';
 import {Readable} from "stream";
 
-let AWS = require('aws-sdk');
-let ftp = require('ftp');
-let fs = require('fs');
-let archiver = require('archiver');
+const AWS = require('aws-sdk');
+const ftp = require('ftp');
+const fs = require('fs');
+const archiver = require('archiver');
 
-let s3 = new AWS.S3();
-let config = new Config();
+const s3 = new AWS.S3();
+const config = new Config();
 
 export async function handler(event) {
     return Promise.all(
-        event.Records.map(record => {
-            let bucket = record.s3.bucket.name;
-            let key = record.s3.object.key;
-            console.log(`Streaming ${bucket}/${key}`);
-
-            if (config.ZipFile) {
-                return streamS3ToLocalZip(bucket, key)
-                    .then(() => ftpConnect(config.FtpHost, config.FtpUser, config.FtpPassword))
-                    .then(ftpClient => streamLocalToFtp(`${key}.zip`, ftpClient))
-            } else {
-                return ftpConnect(config.FtpHost, config.FtpUser, config.FtpPassword)
-                    .then(ftpClient => streamS3FileToFtp(bucket, key, ftpClient))
-            }
-        })
+        event.Records
+            .filter(record => record.s3.object.key.endsWith('csv'))
+            .slice(0, 1)
+            .map(record => {
+                const bucket = record.s3.bucket.name;
+                const key = record.s3.object.key;
+                const today = new Date();
+                const yesterday = `${today.getFullYear()}${today.getMonth() + 1}${today.getDate() - 1}`;
```

Comment:
Is this 100% reliable, particularly at month change boundaries? Is there something like `today.minusDays(1)` or similar we could use for `yesterday`?

Reply:
🤦‍♂️  you're right. will change that.